middlewares/ip_whitelister_test.go

```diff
@@ -0,0 +1,302 @@
+package middlewares
+
+import (
+	"errors"
+	"fmt"
+	"github.com/codegangsta/negroni"
+	"github.com/stretchr/testify/assert"
+	"net"
+	"net/http"
+	"net/http/httptest"
+	"reflect"
+	"testing"
+)
+
+func TestNewIPWhitelister(t *testing.T) {
+	cases := map[string]struct {
+		whitelistStrings   []string
+		expectedWhitelists []*net.IPNet
+		err                error
+	}{
+		"nil whitelist": {
+			whitelistStrings:   nil,
+			expectedWhitelists: nil,
+			err:                errors.New("no whitelists provided"),
+		},
+		"empty whitelist": {
+			whitelistStrings:   []string{},
+			expectedWhitelists: nil,
+			err:                errors.New("no whitelists provided"),
+		},
+		"whitelist containing empty string": {
+			whitelistStrings: []string{
+				"1.2.3.4/24",
+				"",
+				"fe80::/16",
+			},
+			expectedWhitelists: nil,
+			err:                errors.New("parsing CIDR whitelist <nil>: invalid CIDR address: "),
+		},
+		"whitelist containing only an empty string": {
+			whitelistStrings: []string{
+				"",
+			},
+			expectedWhitelists: nil,
+			err:                errors.New("parsing CIDR whitelist <nil>: invalid CIDR address: "),
+		},
+		"whitelist containing an invalid string": {
+			whitelistStrings: []string{
+				"foo",
+			},
+			expectedWhitelists: nil,
+			err:                errors.New("parsing CIDR whitelist foo: invalid CIDR address: foo"),
+		},
+		"IPv4 & IPv6 whitelist": {
+			whitelistStrings: []string{
+				"1.2.3.4/24",
+				"fe80::/16",
+			},
+			expectedWhitelists: []*net.IPNet{
+				{IP: net.IPv4(1, 2, 3, 0), Mask: net.IPv4Mask(255, 255, 255, 0)},
+				{IP: net.ParseIP("fe80::"), Mask: net.IPMask(net.ParseIP("ffff::"))},
+			},
+			err: nil,
+		},
+		"IPv4 only": {
+			whitelistStrings: []string{
+				"127.0.0.1/8",
+			},
+			expectedWhitelists: []*net.IPNet{
+				{IP: net.IPv4(127, 0, 0, 0), Mask: net.IPv4Mask(255, 0, 0, 0)},
+			},
+			err: nil,
+		},
+	}
+
+	for index, e := range cases {
+		t.Run(index, func(t *testing.T) {
+			t.Parallel()
+			whitelister, err := NewIPWhitelister(e.whitelistStrings)
+			if !reflect.DeepEqual(err, e.err) {
+				t.Errorf("expected err: %+v, got: %+v", e.err, err)
+				return
+			}
+			if e.err != nil {
+				// expected error
+				return
+			}
```

Comment:
No, line 80 checks if the returned error-value is correct. It there is an error, and the error was expected, we should not try to continue the test, because -as expected- an error has happened before.
This line ends the test when an error was to be expected.